pkg/operator/controllers/checkers/ingresscertificatechecker/controller_test.go

```diff
@@ -58,7 +58,7 @@ func TestReconcile(t *testing.T) {
 		{
 			name:                 "no default cert set",
 			wantConditionStatus:  operatorv1.ConditionFalse,
-			wantConditionMessage: "ingress has no default certificate set",
+			wantConditionMessage: "missing ingress certificate for cluster with custom domain",
```

Comment:
maybe have a extra test case here too with your new error

Reply:
Interesting! I already asked this question myself before. 

The key thing is that in this test (controller_test.go) we don't really want to test all error cases because the way the controller handles the error is the same for all the errors that the checker returns. I think that we should test just that the error returned by the checker is well used by the controller, no matter which error it is. (so the use case to test in the controller would be: "when the checker returns any error, the controller can handle it properly").

Having in controller_test.go a test case for each of the errors that the checker returns is redundant with the tests of the checker and a change in the prod code will imply touching 2 different unit tests.
 
I believe that all those test cases should be in the checker as it is the one that creates them and there you want to make sure each of them is well created/returned.

What I mentioned about the controller: in the [function](https://github.com/Azure/ARO-RP/blob/master/pkg/operator/controllers/checkers/ingresscertificatechecker/controller.go#L104), it just checks if the error is not nil, so no specific logic with the error. -> we cover all code paths with just one test cases for the error handling.

I changed the name of the test to better reflect that we just want to test that the controller behaves as expected when the checker returns an error, no matter which error. 

Thanks for pointing it out. 
😃